include/boost/geometry/formulas/thomas_first_order_direct.hpp

```diff
@@ -0,0 +1,73 @@
+// Boost.Geometry
+
+// Copyright (c) 2017 Oracle and/or its affiliates.
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_FORMULAS_THOMAS_FIRST_ORDER_DIRECT_HPP
+#define BOOST_GEOMETRY_FORMULAS_THOMAS_FIRST_ORDER_DIRECT_HPP
+
+
+#include <boost/math/constants/constants.hpp>
+
+#include <boost/geometry/core/radius.hpp>
+
+#include <boost/geometry/util/condition.hpp>
+#include <boost/geometry/util/math.hpp>
+
+#include <boost/geometry/formulas/differential_quantities.hpp>
+#include <boost/geometry/formulas/flattening.hpp>
+#include <boost/geometry/formulas/result_direct.hpp>
+#include <boost/geometry/formulas/thomas_direct.hpp>
+
+
+namespace boost { namespace geometry { namespace formula
+{
+
+
+/*!
+\brief The solution of the direct problem of geodesics on latlong coordinates,
+       Forsyth-Andoyer-Lambert type approximation with first order terms.
+\author See
+    - Technical Report: PAUL D. THOMAS, MATHEMATICAL MODELS FOR NAVIGATION SYSTEMS, 1965
+      http://www.dtic.mil/docs/citations/AD0627893
+    - Technical Report: PAUL D. THOMAS, SPHEROIDAL GEODESICS, REFERENCE SYSTEMS, AND LOCAL GEOMETRY, 1970
+      http://www.dtic.mil/docs/citations/AD0703541
+*/
+template <
+    typename CT,
+    bool EnableCoordinates = true,
+    bool EnableReverseAzimuth = false,
+    bool EnableReducedLength = false,
+    bool EnableGeodesicScale = false
+>
+class thomas_first_order_direct
+{
+
+public:
+    typedef result_direct<CT> result_type;
+
+    template <typename T, typename Dist, typename Azi, typename Spheroid>
+    static inline result_type apply(T const& lo1,
+                                    T const& la1,
+                                    Dist const& distance,
+                                    Azi const& azimuth12,
+                                    Spheroid const& spheroid)
+    {   return thomas_direct<
```

Comment:
I propose to implement thomas as one struct taking `int Order` or `bool FirstOrder` as the first or second template parameter, e.g. simply as `thomas_direct`. And then derive both thomas 1st and 2nd order from that if needed. You can implement all of them in one file. If the order flag was struct template parameter you could simply derive from it without the need to implement the interface again.

Reply:
I followed a common interface solution since some formulas rely on it e.g. https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/formulas/gnomonic_intersection.hpp#L39

Now changed it to (the more convenient, I agree) interface with SecondOrder as template parameter and use strategies (in particular formula policies) for gnomonic_intersection in tests https://github.com/boostorg/geometry/pull/489/files#diff-faacdab3bb307b84e30e9bde5ca88591R74 However, this should be fixed in the future. Maybe by just moving formulas policies into formulas?